plugins/inputs/ping/ping.go

```diff
@@ -82,6 +83,20 @@ type Ping struct {
 	Size *int
 }
 
+type roundTripTimeStats struct {
+	min    float64
+	avg    float64
+	max    float64
+	stddev float64
+}
+
+type stats struct {
+	trans int
+	recv  int
+	ttl   int
+	roundTripTimeStats
```

Comment:
If we aren't keeping an array of these round trip stats, why does it need to be yet another type of stats struct?

Reply:
Thanks to that we will have proper number of returned arguments in `checkRoundTripTimeStats` function.

Reply:
If you are only looking at what the functions are returning then that makes sense. However, I do feel this is adding overhead with respect to maintenance of this code.

I've seen you pull functions into a struct before, is it that something you considered here? For example, pulling `checkRoundTripTimeStats` into a Ping type and update the values in that function?

Reply:
I believe that thanks to this change we don't have a lot of unnamed returned arguments flying around so code is more readable (with just a little bit of overhead ofc). 

Indeed, I've pulled some functions into struct few times. But thanks to that these functions could use fields from "main" type (mostly arguments from configuration or types created during initialization). In this case we are talking about metrics which need to be gathered (so completely different thing).

Reply:
Well, I am not convinced :) , but I don't want to hold this up at this point so I will resolve.